### PR TITLE
e2e: Remove explicit timeouts for variable products and analytics tests

### PR DESCRIPTION
**Pull request by mattsherman**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR improves the readability and maintainability of the variable products and analytics e2e tests by removing explicit timeouts.

#### Waiting for data to be saved

Instead of waiting a set amount of time to accommodate the saving of data:

```js
await page.waitForTimeout( 1000 );
```

We wait for the actual network response:

```js
await page.waitForResponse(
  ( response ) =>
	  response.url().includes( '/users/' ) &&
	  response.status() === 200
);
```

#### Avoiding `page.waitForSelector()` and `page.fill()`

By default, `page.fill()` does not require the selector to resolve to a single element. If multiple elements match the selector, the first one is used. This can result in confusing and unexpected behavior. It is recommended that `page.locator().fill()` be used instead. ([Playwright doc](https://playwright.dev/docs/api/class-page#page-fill))

If `page.locator()` is used, then `page.waitForSelector()` does not need to be used. Also, by default, `page.waitForSelector()` does not require the selector to resolve to a single element.  It is recommended that `Locator` objects be used instead. ([Playwright doc](https://playwright.dev/docs/api/class-page#page-wait-for-selector))

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Run e2e tests and verify they reliably pass (if the e2e test check on this PR passes, they did).

Note: To run just the tests modified in this PR locally, you can do the following after setting up your e2e env:

```
pnpm playwright test --config=tests/e2e-pw/playwright.config.js ./tests/e2e-pw/tests/admin-analytics/analytics-overview.spec.js
pnpm playwright test --config=tests/e2e-pw/playwright.config.js ./tests/e2e-pw/tests/merchant/create-variable-product.spec.js
```

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/create-variable-product.spec.js [hunk 1]
```diff
@@ -55,19 +55,28 @@ test.describe.serial( 'Add New Variable Product Page', () => {
...
 
-			await page.waitForSelector( input, { timeout: 1000 } ); // Wait for up to 1 second
-			await page.fill( input, `attr #${ i + 1 }` );
-			await page.fill(
-				`textarea[name="attribute_values[${ i }]"]`,
-				'val1 | val2'
-			);
+			await page
+				.locator(
+					`.woocommerce_attribute_data input[name="attribute_names[${ i }]"]`
```
Comment (mattsherman) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/create-variable-product.spec.js:61]: The selector `input[name="attribute_names[${ i }]"]` matches more than one element in the DOM (there is a hidden input with the same name). So, I added the `.woocommerce_attribute_data` class to make it more specific.

**Review by octaedro — APPROVED** [review]
Thank you @mattsherman for tackling this! 
I would love to see the `page.fill` replacement in the other test. I think we will have enough time to do it during this cycle after finishing the epics.
Great job! 🚀

File: plugins/woocommerce/tests/e2e-pw/tests/admin-analytics/analytics-overview.spec.js [hunk 2]
```diff
@@ -36,7 +36,13 @@ test.describe( 'Analytics pages', () => {
...
 			);
 			await page.click( 'text=Move up' );
-			await page.waitForTimeout( 1000 );
+
+			// wait for the changes to be saved
+			await page.waitForResponse(
+				( response ) =>
+					response.url().includes( '/users/' ) &&
+					response.status() === 200
+			);
```
Comment (octaedro) [inline plugins/woocommerce/tests/e2e-pw/tests/admin-analytics/analytics-overview.spec.js:45]: ♥️

**Comment by mattsherman** [comment]
> I would love to see the `page.fill` replacement in the other test. I think we will have enough time to do it during this cycle after finishing the epics.

Agreed! Let's cycle back to this after taking care of the cycle work.

**Comment by mattsherman** [comment]
> Thank you @mattsherman for tackling this! I would love to see the `page.fill` replacement in the other test. I think we will have enough time to do it during this cycle after finishing the epics. Great job! 🚀

@octaedro Issue created: https://github.com/woocommerce/woocommerce/issues/37458

**Pull request merged**